aewb_logger/aewb_logger_sender.c

```diff
@@ -0,0 +1,239 @@
+#include <stdlib.h>
+#include <unistd.h>
+#include <time.h>
+#include <errno.h>
+#include "aewb_logger_sender.h"
+#include "aewb_logger_types.h"
+
+void get_ip_port_with_envvar_override(const char *in_dest_ip, in_port_t in_dest_port, bool *out_enable, const char **out_dest_ip, in_port_t *out_dest_port)
+{
+    const char *env_aewb_log_en_str = getenv("AEWB_LOG_EN");
+    const char *env_aewb_log_dest_ip_str = getenv("AEWB_LOG_DEST_IP");
+    const char *env_aewb_log_dest_port_str = getenv("AEWB_LOG_DEST_PORT");
+    in_port_t env_aewb_log_dest_port = 0U;
+    
+    if (env_aewb_log_en_str && strcmp(env_aewb_log_en_str, "0") == 0U) {
+        *out_enable = false;
+    }
+    else {
+        *out_enable = true;
+    }
+
+    if (env_aewb_log_dest_ip_str) {
+         // env var ip takes precedence over function arg
+        *out_dest_ip = env_aewb_log_dest_ip_str;
+    }
+    else {
+        *out_dest_ip = in_dest_ip;
+    }
+
+    if (env_aewb_log_dest_port_str) {
+         // env var port takes precedence over function arg
+        errno = 0U;
+        env_aewb_log_dest_port = strtol(env_aewb_log_dest_port_str, NULL, 10);
+        if (errno)
+            env_aewb_log_dest_port = 0U;
+    }
+
+    if (env_aewb_log_dest_port > 0U) {
+        *out_dest_port = env_aewb_log_dest_port;
+    }
+    else { 
+        *out_dest_port = (in_port_t)in_dest_port;
+    }
+}
+
+aewb_logger_sender_state_t *aewb_logger_create_sender(const char *in_dest_ip, in_port_t in_dest_port)
+{
+    const char *final_dest_ip = NULL;
+    in_port_t final_dest_port = {0U};
+    bool final_enable = false;
+    aewb_logger_sender_state_t *p_state = NULL;
+
+    get_ip_port_with_envvar_override(in_dest_ip, in_dest_port, &final_enable, &final_dest_ip, &final_dest_port);
+    if (final_enable == false) {
+        return NULL; // future aewb_logger_send_log() calls will not send anything
```

Comment:
```suggestion
       goto handle_err; // future aewb_logger_send_log() calls will not send anything
```

Reply:
this is not an error case like the rest, i intentionally left it separte.